documentation/Database-Schema-Template.md

```diff
@@ -0,0 +1,75 @@
+With templates versioning we keep previous finalised applications linked to the correct versions of the application templates used. 
+
+* <b>template version</b>
+The `is_current` is set to `'false'` while the version is being created or edited or for all versions that is not the current.  
+The `is_current` is set to `'true'` when the new version of one application template is published. 
+When a new version is added every new application of this application template will be linked to the new version.
+
+<b>To be considered: Should we check if no applications are associated with an existing template and just add changes to the current version instead?</b>
+
+### Visualisation
+The visualisation and questions required in each stage of an application is defined in the application template with the following entities. 
+* <b>template</b>
+Representation of the application template. All nested elements are accessible via joined tables and can be created or queried in the same call using the GraphQL engine.
+The `version_id` is the link with the template version.
+The `template_name` and `code` is to help admin users identifying one template.
+The `current_status` can be `'Draft'`, `'Available'` or `'Disabled'`. The only 'Available' templates are the onesthe in the version flagged as the current one. All other application template are 'Disabled' or 'Draft' (if unfinalised).
+* <b>template stage</b>
+There is one or more stages per application template. Each one is defined as a new record that point to `tenplate_id` with the `number` of this stage (1, 2, 3, ...) and `title` can be anything, the most populars are: `'Screening'`, `'Assessment'`, `'Final Stage'`.
+* <b>template sections</b>
+Sections of the application template that contain some elements. Each section can be associated with a <b>template permission</b> if this section requires a certain type of reviewer to check for responses from an Applicant.
+* <b>template elements</b>
+Elements in the application template always are part of a section, therefor it stores the `section_id` instead of the `template_id`.
+The `code` is associated with the <b>element type plugin</b>, where the definitions of each element will be coming from.
+The order each element is displayed is defined by the `next_element_code` (combined with the `section_id` of the current template version).
+The `visibility_condition` checks for any required previous elements to be answered before this element can be displayed. 
+The `category` is either `'Question'` or `'Information'`. 
+'Questions' elements will require responses from the Applicant and 'Information' elements are only structural or for visualisation. 
+When the element category is 'Question' it also uses `is_required` and `is_editable` to determine its state. 
```

Comment:
Yea, I had a sort of similar though... It's described in the other PR: #15 
I was thinking, do we need a `application_question` table maybe? What I though this `is_editable` would cover is when a question is selected to go back to the applicant for a new response. But if the question is only set in a single place `template_element` like you said we would have to add a dynamic field to define if it's editable. But in this case is even more specific, because only when the question is selected by the consolidator it is editable.

Reply:
Right, I actually answered the `application_question` comment in your other PR.
I think we have different expectations of what the `is_editable` field is to be used for. I always intended it to be a way to questions to become uneditable under certain conditions, such as getting a forced response if any earlier response is one thing, but being enterable if they answered something else. I think you were expecting it to be for marking whether a question is editable due to review status, etc. 

Anyway, regarding how we know if a question can be edited in response to review status -- does [this](https://github.com/openmsupply/application-manager-server/pull/25#discussion_r491757271) answer it?

Reply:
Updated with what we have decided in the meeting on 22-09-2020.